pkg/controller/klusterletaddon/globalproxy_controller.go

```diff
@@ -241,10 +296,49 @@ func getGlobalProxyInInstallConfig(installConfig []byte) (agentv1.ProxyConfig, e
 	if err != nil {
 		return proxyConfig, err
 	}
-	proxyConfig.NoProxy, _, err = unstructured.NestedString(proxyConfigRaw, "proxy", "noProxy")
+
+	noProxy, _, err := unstructured.NestedString(proxyConfigRaw, "proxy", "noProxy")
+	if err != nil {
+		return proxyConfig, err
+	}
+
+	if proxyConfig.HTTPProxy == "" && proxyConfig.HTTPSProxy == "" && noProxy == "" {
+		return proxyConfig, nil
+	}
+
+	if noProxy == "*" {
+		proxyConfig.NoProxy = noProxy
+		return proxyConfig, nil
+	}
+
+	//noProxyList := []string{".cluster.local", ".svc", "localhost", "127.0.0.1"}
```

Comment:
add a doc here on why we need to add pod ip range etc here.

Reply:
done